foreshadow/metrics/internals.py

```diff
@@ -102,3 +106,74 @@ def avg_col_regex(feature, cleaner, mode=min):
     return sum(
         [mode(list_lens) / row_len for list_lens, row_len in matched_lens]
     ) / len(feature)
+
+
+@metric(default_return=0)
+def num_valid(X):
+    """Count the number of valid numbers in an input.
+
+    Args:
+        X (iterable): Input data
+
+    Returns:
+        A proportion of the data that evaluated as an number
+
+    """
+    X = check_series(X)
+    data = ~X.apply(pd.to_numeric, errors="coerce").isnull()
+
+    return float(data.sum()) / data.size
+
+
+@metric(default_return=0)
+def unique_heur(X):
+    """Compute the ratio of unique numbers to the total size of the input.
+
+    Args:
+        X (iterable): Input data
+
+    Returns:
+        1 - the proportion of the data that is unique (ie more unique results \
+            in a number closer to one)
+
+    """
+    X = check_series(X)
+    return 1 - (1.0 * X.nunique() / X.count())
+
+
+@metric(default_return=0)
+def is_numeric(X):
+    """Check if an input is numeric.
+
+    Note:
+        Uses pandas method, `is_numeric_dtype \
+            <pandas.api.types.is_numeric_dtype>` to make determination.
+
+    Args:
+        X (iterable): Input data
+
+    Returns:
+        return True if the input is a numeric type, False otherwise.
+
+    """
+    X = check_series(X)
+    return is_numeric_dtype(X)
+
+
+@metric(default_return=0)
```

Comment:
Told you class-based decorators would be useful ;)

Reply:
haha you were right